src/state/liveTokens/liveTokensStore.ts

```diff
@@ -0,0 +1,271 @@
+    disableCache: true,
+    staleTime: time.seconds(5),
+    setData: ({ data, set }) => {
+      if (data) {
+        set(state => {
+          updateUserAssetsStore(data);
+          return {
+            ...state,
+            tokens: { ...state.tokens, ...data },
+          };
+        });
+      }
+    },
+    paramChangeThrottle: 250,
+    params: {
+      subscribedTokensByRoute: ($, store) => $(store).subscribedTokensByRoute,
+      activeRoute: $ => $(useNavigationStore).activeRoute,
+      currency: $ => $(userAssetsStoreManager).currency,
+    },
+  },
+
+  set => ({
+    ...initialState,
+    addSubscribedTokens({ route, tokenIds }: UpdateSubscribedTokensParams) {
+      set(state => {
+        const { subscribedTokensByRoute } = state;
+
+        // TODO: deduplicate tokenUniqueIds, remove tokens where we only need to fetch the mainnet price
+        let hasChanges = false;
+
+        if (!subscribedTokensByRoute[route]) {
+          subscribedTokensByRoute[route] = {};
+        }
+
+        for (const tokenId of tokenIds) {
+          if (!subscribedTokensByRoute[route][tokenId]) {
+            subscribedTokensByRoute[route][tokenId] = (subscribedTokensByRoute[route][tokenId] ?? 0) + 1;
+            hasChanges = true;
+          }
+        }
+
+        if (!hasChanges) return state;
+
+        return {
+          subscribedTokensByRoute,
+        };
+      });
+    },
+
+    removeSubscribedTokens({ route, tokenIds }: UpdateSubscribedTokensParams) {
+      set(state => {
+        const { subscribedTokensByRoute } = state;
+
+        let hasChanges = false;
+        for (const tokenId of tokenIds) {
+          if (subscribedTokensByRoute[route]?.[tokenId]) {
+            subscribedTokensByRoute[route][tokenId] -= 1;
+            if (subscribedTokensByRoute[route][tokenId] === 0) {
+              delete subscribedTokensByRoute[route][tokenId];
+            }
+            hasChanges = true;
+          }
+        }
+
+        if (!hasChanges) return state;
+
+        // remove routes with no subscriptions
+        if (Object.keys(subscribedTokensByRoute[route]).length === 0) {
+          delete subscribedTokensByRoute[route];
+        }
+
+        return {
+          subscribedTokensByRoute,
+        };
+      });
+    },
+
+    clear() {
+      set({ ...initialState });
+    },
+  })
+);
+
+export function addSubscribedToken({ route, tokenId }: { route: string; tokenId: string }) {
+  useLiveTokensStore.getState().addSubscribedTokens({ route, tokenIds: [tokenId] });
+}
+export function removeSubscribedToken({ route, tokenId }: { route: string; tokenId: string }) {
+  useLiveTokensStore.getState().removeSubscribedTokens({ route, tokenIds: [tokenId] });
+}
+
+export const { addSubscribedTokens, removeSubscribedTokens } = useLiveTokensStore.getState();
+
+export function getBalance({
+  token,
+  balanceAmount,
+  nativeCurrency,
+}: {
+  token: TokenData;
+  balanceAmount: string;
+  nativeCurrency: SupportedCurrencyKey;
+}) {
```

Comment:
At least for exported functions, I think explicit return types are generally good practice.